lib/src/metta/runner/stdlib.metta

```diff
@@ -193,3 +324,338 @@
+(@doc &self
+  (@desc "Returns reference to the current atomspace")
+  (@params ())
+  (@return "Reference to the current atomspace"))
+
+; TODO: get-doc/help! not working for +
+(@doc +
+  (@desc "Sums two numbers")
+  (@params (
+    (@param "Addend")
+    (@param "Augend")))
+  (@return "Sum"))
+
+; TODO: get-doc/help! not working for -
+(@doc -
+  (@desc "Subtracts second argument from first one")
+  (@params (
+    (@param "Minuend")
+    (@param "Deductible")))
+  (@return "Difference"))
+
+; TODO: get-doc/help! not working for *
+(@doc *
+  (@desc "Multiplies two numbers")
+  (@params (
+    (@param "Multiplier")
+    (@param "Multiplicand")))
+  (@return "Product"))
+
+; TODO: get-doc/help! not working for /
+(@doc /
+  (@desc "Divides first argument by second one")
+  (@params (
+    (@param "Dividend")
+    (@param "Divisor")))
+  (@return "Fraction"))
+
+; TODO: get-doc/help! not working for %
+(@doc %
+  (@desc "Modulo operator. It returns remainder of dividing first argument by second argument")
+  (@params (
+    (@param "Dividend")
+    (@param "Divisor")))
+  (@return "Remainder"))
+
+; TODO: get-doc/help! not working for <
+(@doc <
+  (@desc "Less than. Checks if first argument is less than second one")
+  (@params (
+    (@param "First number")
+    (@param "Second number")))
+  (@return "True if first argument is less than second, False - otherwise"))
+
+; TODO: get-doc/help! not working for >
+(@doc >
+  (@desc "Greater than. Checks if first argument is greater than second one")
+  (@params (
+    (@param "First number")
+    (@param "Second number")))
+  (@return "True if first argument is greater than second, False - otherwise"))
+
+; TODO: get-doc/help! not working for <=
+(@doc <=
+  (@desc "Less than or equal. Checks if first argument is less than or equal to second one")
+  (@params (
+    (@param "First number")
+    (@param "Second number")))
+  (@return "True if first argument is less than or equal to second, False - otherwise"))
+
+; TODO: get-doc/help! not working for >=
+(@doc >=
+  (@desc "Greater than or equal. Checks if first argument is greater than or equal to second one")
+  (@params (
+    (@param "First number")
+    (@param "Second number")))
+  (@return "True if first argument is greater than or equal to second, False - otherwise"))
+
+; TODO: get-doc/help! not working for ==
+(@doc ==
+  (@desc "Checks equality for two arguments of the same type")
+  (@params (
+    (@param "First argument")
+    (@param "Second argument")))
+  (@return "Returns True if two arguments are equal, False - otherwise. If arguments are of different type function returns Error currently"))
+
+; TODO: get-doc/help! not working for and
+(@doc and
+  (@desc "Logical conjunction of two arguments")
+  (@params (
+    (@param "First argument")
+    (@param "Second argument")))
+  (@return "Returns True if both arguments are True, False - otherwise"))
+
+; TODO: get-doc/help! not working for or
```

Comment:
@luketpeterson I would like to check it with you. Analysis shows that it doesn't work because when this module is loaded `or` token is replaced by Rust implementation. But when user executes `!(help or)` `or` is replaced by the Python implementation it is the reason why documentation is not returned (Python and Rust implementations do not match). 

In `MettaMod` code I see the following snippet: https://github.com/trueagi-io/hyperon-experimental/blob/5ab9f113dd65ecae4222298af986345430879c9f/lib/src/metta/runner/modules/mod.rs#L96-L100

To me it sounds like one one implementation of the `stdlib` tokens should be loaded: either Rust or Python but looks like in fact we have two of them at the same time. Is it made by intention or it is a bug?

Reply:
If it is a bug and we can fix it then it could fix the documentation behaviour. On the other hand this operation duality can mean that we need separate documentation for Rust grounded atoms and Python grounded atoms.

Reply:
> To me it sounds like one one implementation of the stdlib tokens should be loaded

The python `stdlib` *could* be changed to not load `corelib`, but that's not what it currently does.  Right now it loads `corelib` (which is the Rust stdlib), and overwrites / supersedes certain definitions from corelib with its own definitions.

But I feel like this docs issue is only tangentially related to that question.

My first thought is: If the Python `hyperon/stdlib.py` is providing alternative versions of these operation atoms, it should provide documentation for them.

But my second thought is: Since the changes made a few months ago make the type conversion for grounded atoms much more flexible and add powerful arithmetic & logic ops to `corelib`, can we get rid of these op atoms in `hyperon/stdlib.py` altogether?  Although certainly in a different PR from this one?

Reply:
Yeah, to me these last two changes you are talking about are also a orthogonal to each other. 

One question is about documentation and here I am starting to think that it is more logical to have documentation for the grounded atom inside its implementation like we having grounded atom's type. Thus this documentation is always available from the atom itself. It contradicts to the way I wanted to see grounded atoms in future but on the other hand it is better to have internally aligned representation and switch to another one in one step.

Second question is should we have duplicates of the operations in Python stdlib. The idea is to keep Python stdlib working with native data in a native way. Taking into account calling Python method anyway requires cloning vales I doubt implementing operations on numbers and boolean values in Python makes much sense . Strings serialization is still not implemented thus we cannot fully remove duplicates probably. Thus may be we could get rid of duplicates in Python.